sunpy/net/jsoc/jsoc.py

```diff
@@ -345,121 +343,27 @@ def get(self, jsoc_response, path=None, overwrite=False, progress=True,
-        path = os.path.expanduser(path)
-
-        if downloader is None:
-            downloader = Downloader(max_conn=max_conn, max_total=max_conn)
-
-        # A Results object tracks the number of downloads requested and the
-        # number that have been completed.
-        if results is None:
-            results = Results(lambda _: downloader.stop())
-
-        urls = []
-        for request_id in requestIDs:
-            u = self._request_status(request_id)
-
-            if u.status_code == 200 and u.json()['status'] == '0':
-                for ar in u.json()['data']:
-                    is_file = os.path.isfile(os.path.join(path, ar['filename']))
-                    if overwrite or not is_file:
-                        url_dir = BASE_DL_URL + u.json()['dir'] + '/'
-                        urls.append(urllib.parse.urljoin(url_dir, ar['filename']))
-
-                    else:
-                        print_message = "Skipping download of file {} as it " \
-                                        "has already been downloaded"
-                        print(print_message.format(ar['filename']))
-                        # Add the file on disk to the output
-                        results.map_.update({ar['filename']:{'path':os.path.join(path, ar['filename'])}})
-
-                if progress:
-                    print_message = "{0} URLs found for download. Totalling {1}MB"
-                    print(print_message.format(len(urls), u.json()['size']))
-
+        for block in jsoc_response.query_args:
+            query_string = self._make_recordset(block.get('start_time'),
+                                                block.get('end_time'),
+                                                block.get('series'),
+                                                block.get('wavelength'),
+                                                block.get('segment'))
+            email = block.get('notify', '')
+            protocol = block.get('protocol', 'FITS')
+
+            if not email:
+                raise ValueError("JSOC queries now require a valid email address "
+                                 "before they will be accepted by the server")
             else:
-                if progress:
-                    self.check_request(request_id)
+                client = drms.Client(email=email, verbose=True)
+                response = client.export(query_string, method=method, protocol=protocol)
 
-        if urls:
-            for url in urls:
-                downloader.download(url, callback=results.require([url]),
-                                    errback=lambda x: print(x), path=path)
+            if path is None:
+                path = config.get('downloads', 'download_dir')
+            path = os.path.expanduser(path)
 
-        else:
-            # Make Results think it has finished.
-            results.require([])
-            results.poke()
-
-        return results
+            response.download(path)
```

Comment:
you need to collect all the output here to give the user back a list of filenames or a downloader instance (I don't know how drms.download works)